alacritty/src/config/window.rs

```diff
@@ -74,6 +74,23 @@ impl WindowConfig {
     pub fn set_dynamic_title(&mut self, dynamic_title: bool) {
         self.dynamic_title.0 = dynamic_title;
     }
+
+    #[inline]
+    pub fn set_dimensions(&mut self, dimensions: Dimensions) {
+        self.dimensions = dimensions;
+    }
+
+    #[inline]
+    pub fn dimensions(&self) -> Option<Dimensions> {
+        if self.dimensions.columns.0 != 0
+            && self.dimensions.lines.0 != 0
+            && self.startup_mode != StartupMode::Maximized
```

Comment:
That's a good point, you think we should change that in this PR though? I think on macos/Windows this might actually lead to incorrect behavior. On Linux we call `set_maximized` after startup anyways.

Reply:
let's keep it that way, I don't feel comfortable in changing that. I just now that it'll work on Wayland, but that's it.

Reply:
Oh, can we change the check to `self.startup_mode == StartupMode::Windowed`?

Reply:
Then we wouldn't set the window size correctly for fullscreen applications?

Reply:
Ah, ok.